builtin/logical/transit/path_hmac.go

```diff
@@ -116,70 +136,105 @@ func (b *backend) pathHMACWrite(ctx context.Context, req *logical.Request, d *fr
 		return nil, fmt.Errorf("HMAC key value could not be computed")
 	}
 
-	var hf hash.Hash
+	var hashAlg func() hash.Hash
```

Comment:
This should probably use the keysutil.HashTypeMap stuff we are using elsewhere in `transit`